src/main/java/software/amazon/smithy/gradle/SmithyUtils.java

```diff
@@ -155,17 +157,27 @@ public static File getProjectionOutputDir(Project project) {
     public static void executeCli(Project project, List<String> arguments, FileCollection classpath) {
         FileCollection resolveClasspath = resolveCliClasspath(project, classpath);
         boolean fork = project.getExtensions().getByType(SmithyExtension.class).getFork();
-        project.getLogger().debug("Executing Smithy CLI in a {}: {}; using classpath {}",
+        LOGGER.fine(String.format("Executing Smithy CLI in a %s: %s; using classpath %s",
                                   fork ? "process" : "thread",
                                   String.join(" ", arguments),
-                                  resolveClasspath.getAsPath());
+                                  resolveClasspath.getAsPath()));
         if (fork) {
             executeCliProcess(project, arguments, resolveClasspath);
         } else {
             executeCliThread(arguments, resolveClasspath);
         }
     }
 
+    public static File resolveOutputDirectory(File setOnTask, SmithyExtension extension, Project project) {
```

Comment:
This is now used throughout the Smithy tasks (SmithyBuildJar and SmithyBuild). It is called lazily from each task's getter method for the output directory.